docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
+- Horizontally scalable (capacity and performance)
+- Accessible from a central end point
+- Highly Available
+- Easy to use query language
+- Service accessible through a UI
+- Monitorable service (health, performances and alerts)
+- Ability to correlate logs with KPIs and alerts
+- Multitenancy for audit logs
+- Performances (TBD)
+
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
```

Comment:
Is this log-source RBAC really required? I mean, will a Storage Admin really go down into the container logs if he can't do more than managing the CRs we expose for him?
The thing is, I'm really not sure this feature (RBAC) is available (neither in Loki nor ES).

Reply:
Don't want to have any kind of RBAC for logs, just saying that only users having the Platform and Storage Admin role will use the service. Other types of users will never need to access those logs. This is how the US are written in the service map

Reply:
Oh yes right, you define what a role does, but it doesn't imply another can't do it per-se. It's a bit confusing, especially if you need to derive RBAC rules from those stories.